cv::ocl not found when building wasm bindings.cpp

##### System information (version)
- OpenCV: head of master (`12cd3b55ecf9e261620356c5bea2df0f8ec47bf5`)
- Operating system: Nixos

##### Detailed description

Building the wasm version of OpenCV fails with the following error:

```
Executing: ['make', '-j', '4', 'opencv.js']
[  6%] Built target zlib
[ 32%] Built target libprotobuf
[ 52%] Built target opencv_core
[ 70%] Built target opencv_imgproc
[ 72%] Built target opencv_objdetect
[ 77%] Built target opencv_video
[ 98%] Built target opencv_dnn
[ 98%] Generating bindings.cpp
Scanning dependencies of target opencv_js
[100%] Building CXX object modules/js/CMakeFiles/opencv_js.dir/bindings.cpp.o
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4024:50: error: no member named
      'ocl' in namespace 'cv'; did you mean 'ogl'?
    constant("SVMCapabilities_SVM_ATOMICS", +cv::ocl::svm::SVMCapabilities::SVM_ATOMICS);
                                             ~~~~^~~
                                                 ogl
/home/sidharth/clones/opencvjs/opencv/modules/core/include/opencv2/core/base.hpp:708:11: note: 'ogl'
      declared here
namespace ogl
          ^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4024:55: error: no member named
      'svm' in namespace 'cv::ogl'
    constant("SVMCapabilities_SVM_ATOMICS", +cv::ocl::svm::SVMCapabilities::SVM_ATOMICS);
                                             ~~~~~~~~~^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4026:62: error: no member named
      'ocl' in namespace 'cv'; did you mean 'ogl'?
    constant("SVMCapabilities_SVM_COARSE_GRAIN_BUFFER", +cv::ocl::svm::SVMCapabilities::SVM_COARS...
                                                         ~~~~^~~
                                                             ogl
/home/sidharth/clones/opencvjs/opencv/modules/core/include/opencv2/core/base.hpp:708:11: note: 'ogl'
      declared here
namespace ogl
          ^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4026:67: error: no member named
      'svm' in namespace 'cv::ogl'
    constant("SVMCapabilities_SVM_COARSE_GRAIN_BUFFER", +cv::ocl::svm::SVMCapabilities::SVM_COARS...
                                                         ~~~~~~~~~^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4028:60: error: no member named
      'ocl' in namespace 'cv'; did you mean 'ogl'?
    constant("SVMCapabilities_SVM_FINE_GRAIN_BUFFER", +cv::ocl::svm::SVMCapabilities::SVM_FINE_GR...
                                                       ~~~~^~~
                                                           ogl
/home/sidharth/clones/opencvjs/opencv/modules/core/include/opencv2/core/base.hpp:708:11: note: 'ogl'
      declared here
namespace ogl
          ^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4028:65: error: no member named
      'svm' in namespace 'cv::ogl'
    constant("SVMCapabilities_SVM_FINE_GRAIN_BUFFER", +cv::ocl::svm::SVMCapabilities::SVM_FINE_GR...
                                                       ~~~~~~~~~^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4030:60: error: no member named
      'ocl' in namespace 'cv'; did you mean 'ogl'?
    constant("SVMCapabilities_SVM_FINE_GRAIN_SYSTEM", +cv::ocl::svm::SVMCapabilities::SVM_FINE_GR...
                                                       ~~~~^~~
                                                           ogl
/home/sidharth/clones/opencvjs/opencv/modules/core/include/opencv2/core/base.hpp:708:11: note: 'ogl'
      declared here
namespace ogl
          ^
/home/sidharth/clones/opencvjs/opencv/build_wasm/modules/js/bindings.cpp:4030:65: error: no member named
      'svm' in namespace 'cv::ogl'
    constant("SVMCapabilities_SVM_FINE_GRAIN_SYSTEM", +cv::ocl::svm::SVMCapabilities::SVM_FINE_GR...
                                                       ~~~~~~~~~^
8 errors generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
```

##### Steps to reproduce

```
python ./platforms/js/build_js.py build_wasm --build_wasm --emscripten_dir=<location of enscripten>
```

--- a/modules/js/CMakeLists.txt
+++ b/modules/js/CMakeLists.txt
@@ -32,6 +32,7 @@ endforeach(m)
 # header blacklist
 ocv_list_filterout(opencv_hdrs "modules/.*.h$")
 ocv_list_filterout(opencv_hdrs "modules/core/.*/cuda")
+ocv_list_filterout(opencv_hdrs "modules/core/.*/opencl")
 ocv_list_filterout(opencv_hdrs "modules/core/include/opencv2/core/opengl.hpp")
 ocv_list_filterout(opencv_hdrs "modules/core/include/opencv2/core/ocl.hpp")
 ocv_list_filterout(opencv_hdrs "modules/cuda.*")

